File: Semester1/B-CPE-110/pushswap/bonus/parsing.c
/*
** EPITECH PROJECT, 2022
** bonus
** File description:
** parsing
*/

#include <stdlib.h>
#include "include/my_struct.h"
#include "include/my.h"

int *get_args(int argc, char *argv[])
{
    int args_nb = 3;
    char *args_list[] = {"-v", "-vT", "-vG"};
    int *args = malloc(sizeof(int) * args_nb);

    if (argc >= 2) {
        for (int i = 0; i < args_nb; i++) {
            args[i] = (!my_strcmp(argv[1], args_list[i]) && !args[i]);
        }
    }
    return (args);
}

File: Semester4/B-MAT-400/204ducks/parsing.py
from sys import argv


def parsing():
    argc = len(argv)
    if argc != 2:
        exit(84)
    try:
        a = float(argv[1])
    except:
        exit(84)
    if a < 0 or a > 2.5:
        exit(84)
    return a
